src/sknnr/_base.py

```diff
@@ -104,33 +120,9 @@ def _set_fitted_transformer(self, X, y):
         """Fit and store the transformer."""
         self.transformer_ = self._get_transformer().fit(X, y)
 
-    @property
-    def feature_names_in_(self):
-        return self.transformer_.feature_names_in_
-
-    @property
-    def n_features_in_(self):
-        return self.transformer_.n_features_in_
-
-    def _check_feature_names(self, X, *, reset):
-        """Override BaseEstimator._check_feature_names to prevent errors.
-
-        This check would fail during fitting because `feature_names_in_` stores original
-        names while X contains transformed names. We instead rely on the transformer to
-        check feature names and warn or raise for mismatches.
-        """
-        return
-
-    def _check_n_features(self, X, *, reset):
-        """Override BaseEstimator._check_n_features to prevent errors.
-
-        See _check_feature_names.
-        """
-        return
-
     def fit(self, X, y):
         """Fit using transformed feature data."""
-        self._validate_data(X, y, force_all_finite=True, multi_output=True)
+        _validate_data(self, X, y, ensure_all_finite=True, multi_output=True)
```

Comment:
Check my logic on this one.  It looks like `ensure_all_finite` keyword was introduced in 1.6, so if the user is < 1.6.0, this will use `BaseEstimator._validate_data` which doesn't have support for that keyword, correct?  Do we need to change the keyword from `ensure_all_finite` to `force_all_finite` within the wrapper function?

Reply:
Great catch! I'll fix this in the wrapper. 

This makes me think that maybe we do need a way to easily test in < 1.6.0, to avoid more issues like this in the future... I'll give that some more thought, but will probably defer for a future PR.